### DISCUSSION TITLE
.align and .balign?

### DISCUSSION THREAD

**Discussion by utmandrew** [mortbopet/Ripes#109] [Q&A] [answered]
Could I get an example of how to use the alignment directives (if they're supported)? I see examples of how to print strings, bytes, etc. but didn't see a memory alignment example that uses .align or .balign from the RISC assembly manual. 

Specifically:
```
.data
array1:     .byte      97, 98, 99
.align      4
integer:    .word      42
```

Entering the code above puts a syntax warning under the .align line, so I don't think it's recognized. When run, any directives after the align are ignored.

Thanks!

**Comment by mortbopet** [comment, marked as answer]
Which version of Ripes are you using? For me, using the most recent (v2.2.0), no syntax warnings are present and the data is emitted as expected:
![image](https://user-images.githubusercontent.com/16338943/128072147-b1ed0406-c25a-4310-b91a-4a51e218b68f.png)

**Reply by utmandrew** [reply]
Ripes-2.2.1-linux-x86_64.AppImage -- I'll try an update. Thank you.

Update: 2.2.2 works fine. Thanks again!

**Comment by mortbopet** [comment]
The issue might be that `.balign` is not implemented in the Ripes assembler - I only checked the `.align`, as posted in your code. Unfortunately, only a subset of the directives supported listed in https://github.com/riscv/riscv-asm-manual/blob/master/riscv-asm.md are supported in the built-in Ripes assembler. Supporting more directives is mostly a matter of them being implemented, and not so much a matter of complexity, see: https://github.com/mortbopet/Ripes/blob/master/src/assembler/gnudirectives.cpp
